contrib/pylightning/lightning/lightning.py

```diff
@@ -25,26 +25,23 @@ def _writeobj(sock, obj):
         s = json.dumps(obj)
         sock.sendall(bytearray(s, 'UTF-8'))
 
-    def _readobj(self, sock):
-        buff = b''
+    def _readobj(self, sock, buff=b''):
```

Comment:
Please just make `buff` a member of `UnixDomainSocketRpc`, passing it out and back in is not idiomatic and will cause misuse.

Reply:
No, that *doesnt' work* (and took me a few hours debugging!).  It's simple to change if you want to see it yourself.

When the daemon restarts, for example, we can read half a reply, then fail.  Similarly if we use timeouts in our testing.  That half-reply is stored in self.buff.  Now we try to use the same UnixDomainSocketRPC for another command, and *append* the result to self.buff and get garbage.

And "passing back" is AFAICT the idiomatic way to do pass-by-reference in Python?

(We really want to make it a member of 'sock', so it gets cleared correctly when sock is recreated.  Though even better would be to fix pylightning not to reopen unless necessary?)